Enhancement: Added Kafka OIDC + azure.identity independent producer/consumer examples.

Pilipets opened the pull request:
There were no examples for Kafka OIDC and accessing the Azure event hub without the azure identity library.

This pull request introduces new examples of producer/consumer to address the above issues:
1. OIDC SASL_SSL - `get_oidc_config`
2. SASL_SSL with custom token refresh callback without azure.identity dependency - `get_cb_config`

The current PR is a draft one - let me know if you find it useful.
Then I will create `README.md`, will write code in more OOP-like manner and add more comments for the used concepts.

More context one can gather from the [issue](https://github.com/Azure/azure-event-hubs-for-kafka/issues/223) I created recently.
Examples are based on the existing code in tutorials for [Python-oauth](https://github.com/Azure/azure-event-hubs-for-kafka/tree/master/tutorials/oauth/python).

CC @hmlam

Pilipets committed:
Extended Python Kafka oauth examples with opaque and oidc examples:
- Added an option to run producer, consumer with --mode=[oidc, azure, opaque]
- oidc corresponds to KIP-768 and decoding JWT with the usage of exp claim
- opaque utilizes expires_in field without azure.identity dependency

Signed-off-by: Hlib Pylypets <pilipets.gleb@gmail.com>

Pilipets commented:
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

hmlam reviewed (commented):
Just some house-keeping comments
- since it is OIDC sample, we can probably use the folder "oidc" (or OpenIdConnect) as oppose to more_example.
- sample will need to have the default comment header for licenses etc. example can be taken from other samples.

hpylypets-ua commented:
@hmlam, sorry for the delay in response.

1. Added license headers as you requested.
2. Added config_utils that provides producer configs for the azure, oidc, opaque modes.
3. Extended producer, consumer functionality to be able to run those with --mode which is one of azure, oidc, opaque.

I didn't want to duplicate the same producer, consumer logic for oidc, opaque folders, so needed to change the structure of producer and consumer.
I preserved all the comments and added new ones for oidc, opaque modes.

No need to change README.md because both python scripts can be used as before + with newly added functionality.

Pilipets committed:
Updated comments

Pilipets commented:
@hmlam, sorry for the delay in response.

I didn't want to duplicate the same producer/consumer logic for `oidc`, `opaque` folders, so decided to modify the structure of producer/consumer and extend the usage functionality.

1. Added license headers as you requested.
2. Added `config_utils.py` that provides producer configs for three modes - `azure`, `oidc`, `opaque`.
3. Extended producer, consumer functionality to be able to run those with `--mode` which is one of above mentioned modes.
4. No need to change README.md because both Python scripts can be used as before. Therefore, backward compatibility is maintained, and newly added functionality is optional.
5. Tested all the changes manually.
6.  I preserved all the comments and added new ones for oidc, opaque modes:
    - oidc corresponds to KIP-768 and decoding JWT with the usage of `exp` claim
    - opaque utilizes `expires_in` field and network requests without `azure.identity` dependency
